src/librustc_codegen_llvm/debuginfo/mod.rs

```diff
@@ -224,8 +224,37 @@ impl DebugInfoBuilderMethods<'tcx> for Builder<'a, 'll, 'tcx> {
         gdb::insert_reference_to_gdb_debug_scripts_section_global(self)
     }
 
-    fn set_value_name(&mut self, value: &'ll Value, name: &str) {
-        let cname = SmallCStr::new(name);
+    fn set_var_name(&mut self, value: &'ll Value, name: impl ToString) {
+        // Avoid wasting time if LLVM value names aren't even enabled.
+        if self.sess().fewer_names() {
+            return;
+        }
+
+        // Only function parameters and instructions are local to a function,
+        // don't change the name of anything else (e.g. globals).
+        let param_or_inst = unsafe {
+            llvm::LLVMIsAArgument(value).is_some() ||
+            llvm::LLVMIsAInstruction(value).is_some()
+        };
+        if !param_or_inst {
+            return;
```

Comment:
My first instinct would be to make this a `bug!()`, not a silent nop. Are there any callers (now or after more debuginfo work) that might good reason need to pass in something other than an instruction or argument?

Reply:
Well, yeah, `Constant`s (which would be a nop), or, worse, `GlobalValue`s (functions/statics), which would get their mangled names replaced.

That is, I'm pretty sure `let x = foo as fn();` and `let y = &STATIC;` would both allow accidental renaming of `foo`'s or `STATIC`'s symbol names without this `return`.

Reply:
Ah, that makes sense. Well, silent nop it is then.